src/modules/exchangeProvider.js

```diff
@@ -109,7 +117,11 @@ const onOrderFill = function(tokenBuyer, tokenSeller, buyTokenOrderId, sellToken
     store.dispatch(refreshExchange());
     store.dispatch(refreshOrders());
     const userAccount = store.getState().web3Connect.userAccount;
-    store.dispatch(processNewTrade('OrderFill', userAccount, this));
+    if (userAccount === tokenBuyer || userAccount === tokenSeller) {
+        store.dispatch(processNewTrade('OrderFill', userAccount, this));
+    } else {
+        processNewTrade('OrderFill', userAccount, this)
+    }
```

Comment:
Does it work? Have you tried it?
We want to add two trades if maker and taker are the same.
So something like this should work?
```js
if (userAccount === tokenBuyer ) {
   store.dispatch(processNewTrade('OrderFill', userAccount, this));
}

if( userAccount === tokenSeller) {
   store.dispatch(processNewTrade('OrderFill', userAccount, this));
}
```
(not sure about how you differentiate buy/sell side when parsing, you might need to pass some extra params?)